Port parallel execution of rpc tests

sickpig opened the pull request:
With this PR on my local machine it took 353 sec to run all the enabled tests, it usually took 3/4 times more time, see:

```
ALL                            | True   | 1178 s (accumulated)
Runtime: 353 s
```
While at it I've taken care of issue #584 (tests failed if `netstat` not installed).

This is the list of commits included in the PR. 
af49f52 [qa] pull-tester: Fix assertion and check for run_parallel (MarcoFalke)
62b9bd5 [qa] Adapt BU final test summary to the parallel execution (sickpig)
7d21dcf [qa] Print debug info on stdout rather than stderr (sickpig)
780efde [qa] rpc-tests: Apply random offset to portseed (MarcoFalke)
5abfdea [qa] test_framework: Append portseed to tmpdir (MarcoFalke)
1494e8e [qa] Add option --portseed to test_framework (MarcoFalke)
758562c [qa] pull-tester: Run rpc test in parallel (MarcoFalke)
ca92b30 [qa] test_framework: Exit when tmpdir exists (MarcoFalke)
ca00a3d [qa] Refactor test_framework and pull tester (MarcoFalke)
9cbe296 [qa] Stop other nodes, even when one fails to stop (MarcoFalke)
0b53f29 Autofind rpc tests --srcdir (jonasschnelli)

Commit:
[qa] Stop other nodes, even when one fails to stop

jonasschnelli committed:
Autofind rpc tests --srcdir

Commit:
[qa] Add option --portseed to test_framework

Commit:
[qa] test_framework: Append portseed to tmpdir

This makes it possible to specify a tmpdir while running tests in
parallel

Commit:
[qa] rpc-tests: Apply random offset to portseed

This helps to skip over resources, which are blocked by regtest bitcoind
zombie nodes

Github-Pull: #9098
Rebased-From: fab0f07dec8d6e21ab70843fdce101f1703588fd

Commit:
[qa] test_framework: Exit when tmpdir exists

Commit:
[qa] Refactor test_framework and pull tester

* log to stdout
* increase range for p2p and rpc ports
* Stop nodes on CTRL+C

(partial cherry pick, avoid to apply on rpc-tests.py)

Commit:
[qa] pull-tester: Run rpc test in parallel

manual cherry pick for Core fa494dec

sickpig committed:
[qa] Print debug info on stdout rather than stderr

excessive.py and validateblocktemplate.py use to print
additional info on stderr. After we merged the machinery
to run tests in parallel having a test writing on stderr
means marking the test as failing even if successful.

sickpig committed:
[qa] Adapt BU final test summary to the parallel execution

Commit 758562c (parallel execution) introduce also a new way
to summarize test results. With this commit we removed the
old style report modulo number of passed/failed/skipped/disabled
tests.

Commit:
[qa] pull-tester: Fix assertion and check for run_parallel

(manual cherry-pick)

awemany commented on qa/rpc-tests/test_framework/test_framework.py:
```diff
@@ -159,6 +159,8 @@ def main(self,argsOverride=None,bitcoinConfDict=None,wallets=None):
         random.seed(self.randomseed)
         print("Random seed: %s" % self.randomseed)
 
+        self.options.tmpdir += '/' + str(self.options.port_seed)
```
Nit: This goes back to the original commit: But shouldn't this preferably be a call to os.path.join(..) ?

sickpig replied:
@awemany you are right `os.path.join(..) ` is definitely better and a portale way to handle file path

sickpig replied:
now that I look at the qa python tests code simple string concatenation is used all over the place, wonder if the test suites has never run on windows...

sickpig committed:
[qa] Print debug info on stdout rather than stderr

This is the same as `8106a12` applied to `wallet.py`
and `buip055.py`

sickpig committed:
Use `os.path.join` rather than string concatenation

This would guarantee to generate files path that
are valid in every kind of OS you are going to run
the tests.

Unfortunately it seems there're a lot of other places
where string concatenation is used. I think we should
fix all of them if we want the tests suite to be portable.

sickpig commented:
rebased. 

@ftrader it would be great if you could have a look at this one :smile:

ftrader commented:
@sickpig : I've had a brief look at the updated script and I think the options I introduced before are still there, so it looks ok to me.

sickpig commented:
@ftrader thank you for reviewing this PR. 
Taking out of WIP. I think it is ready to be merged.
